### PR TITLE
Focus Mode Visual Feedback: Pass Classes

### PR DESCRIPTION
**Pull request by jonawals**
This PR contains the pass class files for the Editor Mode Visual Feedback system necessary to implement the Focus Mode visual feedback effects (the rest of the work will come in subsequent PRs).

### CODE REVIEW

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeOutlinePass.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        void EditorModeOutlinePass::FrameBeginInternal(FramePrepareParams params)
+        {
+            SetSrgConstants();
+            EditorModeFeedbackPassBase::FrameBeginInternal(params);
+        }
+
+        bool EditorModeOutlinePass::IsEnabled() const
+        {
+            return true;
+        }
```
Comment (antonmic) [inline Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeOutlinePass.cpp:53]: Same, remove this override

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeTintPass.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        void EditorModeTintPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            SetSrgConstants();
+            EditorModeFeedbackPassBase::FrameBeginInternal(params);
+        }
+
+        bool EditorModeTintPass::IsEnabled() const
+        {
+            return true;
+        }
```
Comment (antonmic) [inline Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeTintPass.cpp:51]: Same, remove

**Review by antonmic — CHANGES_REQUESTED** [review]
Looks good, but some overrides should be removed, otherwise proper pass functionality will be impaired surrounding enabled flags set in JSON

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeBlurPass.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        void EditorModeBlurPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            SetSrgConstants();
+            EditorModeFeedbackPassBase::FrameBeginInternal(params);
+        }
+
+        bool EditorModeBlurPass::IsEnabled() const
+        {
+            return true;
+        }
```
[the remark this answers is not in the document]
Comment (jonawals) [reply Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeBlurPass.cpp:50]: Fixed 👍

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeFeedbackParentPass.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        void EditorModeFeedbackParentPass::InitializeInternal()
+        {
+            ParentPass::InitializeInternal();
+        }
+
+        void EditorModeFeedbackParentPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            ParentPass::FrameBeginInternal(params);
+        }
```
[the remark this answers is not in the document]
Comment (jonawals) [reply Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeFeedbackParentPass.cpp:40]: Fixed 👍

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeFeedbackPassBase.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+#include <PostProcessing/EditorModeFeedbackPassBase.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<EditorModeFeedbackPassBase> EditorModeFeedbackPassBase::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<EditorModeFeedbackPassBase> pass = aznew EditorModeFeedbackPassBase(descriptor);
```
[the remark this answers is not in the document]
Comment (jonawals) [reply Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeFeedbackPassBase.cpp:17]: I'm inheriting the parent `FullscreenTrianglePass` class constructors like so: 

```
using AZ::RPI::FullscreenTrianglePass::FullscreenTrianglePass;
```

File: Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeOutlinePass.cpp (hunk 1 above)
Comment (jonawals) [reply Gems/Atom/Feature/Common/Code/Source/PostProcessing/EditorModeOutlinePass.cpp:53]: Fixed 👍

**Pull request closed without merging**